mono/utils/mono-state.c

```diff
@@ -0,0 +1,476 @@
+/**
+ * \file
+ * Support for verbose unmanaged crash dumps
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+#include <config.h>
+#include <glib.h>
+#include <mono/utils/json.h>
+#include <mono/utils/mono-state.h>
+#include <mono/metadata/object-internals.h>
+
+extern GCStats mono_gc_stats;
+
+// For AOT mode
+#include <mono/mini/mini-runtime.h>
+
+#ifdef TARGET_OSX
+#include <mach/mach.h>
+#include <mach/task_info.h>
+#endif
+
+#define MONO_MAX_SUMMARY_LEN 900
+static JsonWriter writer;
+static GString static_gstr;
+static char output_dump_str [MONO_MAX_SUMMARY_LEN];
+
+static void mono_json_writer_init_static (void) {
+	static_gstr.len = 0;
```

Comment:
None of this is needed, if the data is only used once. It is needed if used more than once.

Reply:
I'm not comfortable saying this will never need to be reentrant / called more than once. I'd rather leave the init if it's all the same.

Reply:
For reentrance you'd want to pass around explicit context but ok.